python/lsst/daf/butler/core/datastore.py

```diff
@@ -80,17 +81,18 @@ class DatastoreValidationError(ValidationError):
 
 @dataclasses.dataclass
 class DatastoreRecordData:
-    """A struct that represents a tabular data export from one or more
-    datastores.
+    """A struct that represents a tabular data export from a single
+    datastore.
     """
 
-    locations: Dict[str, List[DatasetIdRef]] = dataclasses.field(default_factory=lambda: defaultdict(list))
-    """Mapping from datastore name to the datasets in that datastore.
+    refs: List[DatasetRef] = dataclasses.field(default_factory=list)
```

Comment:
I think there was some API that did not accept `DatasetIdRef`, but I will re-check.

Reply:
Not quite that thing, but `BridgeManager.findDatastores()` only accepts true DatasetRef, which forced me to change `export_records` to accept DatasetRefs as well, and that led to changing this structure (though it was not really needed here). Maybe the right thing to do is to change `findDatastores()` to accept DatasetIdRef (or DatasetId)? Bridge API actually uses DatasetRefs extensively, not sure it will be an easy change.

Reply:
I changed `findDatastores()` to accept `DatasetIdRef`, it did not trigger any changes elsewhere.